src/provisioning/api.md

```diff
@@ -0,0 +1,167 @@
+Provisioning API for matrix-hookshot
```

Comment:
can we get error codes so the provisoner can determine what the failure was without parsing error messages?

Reply:
Yep, just needs an addition to the ApiError class

Reply:
Done!